Helper/Order.php

```diff
@@ -660,10 +645,12 @@ public function saveUpdateOrder($reference, $frontend = true, $bolt_trace_id = n
             $this->deleteRedundantQuotes($quote);
         }
 
-        // update order payment transactions
-        $this->updateOrderPayment($order, $transaction);
-
-        if ($frontend) {
+        if (Hook::$fromBolt) {
+            // if called from hook update order payment transactions
+            $this->updateOrderPayment($order, $transaction);
+        } else {
+            // if called from the store controller return quote and order
```

Comment:
Just out of curiosity, why is that? shouldn't we attach payment even if it's still pending?

Reply:
@daisy1754 
If, for some reason, event handler would fail from the frontend call, e.g. totals mismatch, the error would propagate to the end user, breaking the checkout flow, showing a message, etc. It is not necessary, and affects both user and merchant experience with Bolt.
This way, If the payment was successful, the order is saved with the default, pending, status, it is shown in customer dashboard if opened, browser gets redirected to the success page as expected and  the payment update is postponed until the first hook is received (30 seconds?).
If the hook fails due to mismatch, it will fail early, on the first pending hook request, and we will be the first to know. The order will be put ON HOLD with a total mismatch message recorded.